plugins/woocommerce/src/Internal/DataStores/Orders/LegacyDataHandler.php

```diff
@@ -372,13 +433,119 @@ private function order_meta_to_array( \WC_Order &$order ): array {
 	 * @return string[] Property names.
 	 */
 	private function get_order_base_props(): array {
-		return array_column(
-			call_user_func_array(
-				'array_merge',
-				array_values( $this->data_store->get_all_order_column_mappings() )
-			),
-			'name'
-		);
+		$base_props = array();
+
+		foreach ( $this->data_store->get_all_order_column_mappings() as $mapping ) {
+			$base_props = array_merge( $base_props, array_column( $mapping, 'name' ) );
+		}
+
+		return $base_props;
+	}
+
+	/**
+	 * Filters a set of HPOS row updates to those matching a specific set of order properties.
+	 * Hooked onto 'woocommerce_orders_table_datastore_db_rows_for_order'.
```

Comment:
```suggestion
	 * Called via the `woocommerce_orders_table_datastore_db_rows_for_order` filter in `backfill_order_to_datastore`.
```

Reply:
Thanks for the suggestion @lsinger. Commited in de2402aa1559a405aeb6a970dc9e6b8431dde618.